packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -698,11 +698,24 @@ describe('<TreeItem />', () => {
           expect(getByTestId('one')).to.have.attribute('aria-selected', 'false');
           fireEvent.keyDown(document.activeElement, { key: ' ' });
           expect(getByTestId('one')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('one')).to.have.class('Mui-selected');
```

Comment:
```suggestion
          expect(getByTestId('one')).to.have.class(classes.selected);
```

so that it doesn't break when we change these. The fact that these classnames have a stable class is subject of other tests.

Reply:
I agree with the classes.selected - I will change if you agree with me below.

As for noops that makes sense.

Couple of questions :

Given the event check is it necessary for further checks on space and enter ?

```
const handleKeyDown = event => {
    ...
    if (event.altKey || event.currentTarget !== event.target) {
      return;
    }

    switch (key) {
      case ' ':
        if (nodeRef.current === event.currentTarget) {
...
      case 'Enter':
        if (nodeRef.current === event.currentTarget && expandable) {

```

What default action are we preventing ?

Reply:
> What default action are we preventing ?

In general I'm fairly certain you should do this anyway if you you handled a certain keypress. But just because I'm not aware of use cases where keystroke triggers multiple actions.

Specifically here: `Space` triggers scroll; `Enter` triggers ? not sure about that one. Would leave it to be safe.

Reply:
ArrowDown, ArrowUp, Home, End select and focus.  Space and A select only.

So I assume for the first 4 we prevent default regardless of disableSelection and no change there.  

```
case ' ':
        if (nodeRef.current === event.currentTarget) {
          if (multiSelect && event.shiftKey) {
            flag = selectRange(event, { end: nodeId });
          } else if (multiSelect) {
            flag = selectNode(event, nodeId, true);
          } else {
            flag = selectNode(event, nodeId);
          }
        }
        event.stopPropagation();
        break;
..
default:
        if (key === '*') {
          expandAllSiblings(event, nodeId);
          flag = true;
        } else if (multiSelect && ctrlPressed && key.toLowerCase() === 'a') {
          flag = selectAllNodes(event);
```

and change each of the select... functions to return true ?

Reply:
New commit does as I suggested.